src/main/java/io/cdap/plugin/http/sink/batch/HTTPRecordWriter.java

```diff
@@ -74,28 +76,36 @@ public class HTTPRecordWriter extends RecordWriter<StructuredRecord, StructuredR
   public void write(StructuredRecord input, StructuredRecord unused) throws IOException {
     String message = null;
     if (config.getMethod().equals("POST") || config.getMethod().equals("PUT")) {
-      if (config.getMessageFormat().equals("JSON")) {
-        message = StructuredRecordStringConverter.toJsonString(input);
-        contentType = "application/json";
-      } else if (config.getMessageFormat().equals("Form")) {
-        message = createFormMessage(input);
-        contentType = " application/x-www-form-urlencoded";
-      } else if (config.getMessageFormat().equals("Custom")) {
-        message = createCustomMessage(config.getBody(), input);
-        contentType = " text/plain";
+      switch (config.getMessageFormat()) {
+        case "JSON":
+          message = StructuredRecordStringConverter.toJsonString(input);
+          contentType = "application/json";
+          break;
+        case "Form":
+          message = createFormMessage(input);
+          contentType = "application/x-www-form-urlencoded";
+          break;
+        case "Custom":
+          message = createCustomMessage(config.getBody(), input);
+          contentType = "text/plain";
+          break;
+        default:
+          throw new IllegalArgumentException("Invalid message format.");
       }
-      messages.append(message).append(config.getDelimiterForMessages());
+      messageBuffer.add(message);
     }
-    StringTokenizer tokens = new StringTokenizer(messages.toString().trim(), config.getDelimiterForMessages());
-    if (config.getBatchSize() == 1 || tokens.countTokens() == config.getBatchSize()) {
+    int batchSize = config.getBatchSize();
+    if (messageBuffer.size() == batchSize) {
+      flushMessageBuffer();
```

Comment:
with a method named 'flush', I would expect it to actually write to the http service. It's better to move the executeHTTPService() call into the flush method.

This would also remove a little duplicate code in the close() method.

Reply:
Updated, flush method is now responsible for calling the http service